ui/src/app/shared/components/input-filter.tsx

```diff
@@ -8,68 +8,60 @@ interface InputProps {
     onChange: (input: string) => void;
 }
 
-interface InputState {
-    value: string;
-    localCache: string[];
-    error?: Error;
-}
+export function InputFilter(props: InputProps) {
+    const [value, setValue] = useState(props.value);
+    const [localCache, setLocalCache] = useState((localStorage.getItem(props.name + '_inputs') || '').split(',').filter(item => item !== ''));
```

Comment:
ts infers the type of `localCache` correctly, so I omitted type specification. (`useState<string[]>`)
but at the same time, I think it also looks good to specify the type for readability.
which should I choose 😟

Reply:
yea [same as your previous PR](https://github.com/argoproj/argo-workflows/pull/11803#discussion_r1323485883), I think when the inference is correct and straightforward (`string` in this case), there's no need to annotate it.

I also have experience working in ML-family languages like OCaml, which are even stronger typed and where the compiler infers the types of the entire program (so there are _no_ type annotations at all). On the other spectrum would be something like Java, which requires on type annotations on almost everything and feels very verbose as a result (despite having weaker guarantees than OCaml et al).

For TS, I prefer to have strict typing on, which we'll be able to do in this codebase once there is some progress on decoupling `argo-ui` (see https://github.com/argoproj/argo-ui/issues/453). With strict typing, if a type can't be correctly inferred, the compiler will tell you (right now I believe it only has a warning).

Reply:
I see. unless it's a complex type, I'll try to omit the type annotations where TS correctly infers. thanks for detailed description.😄 

and I had no idea about argo-ui! Thank you for letting me know. I'm really hopeful for the progress in decoupling. It sounds it could bring great improvements.